### PR TITLE
Issue/876 - Shorten description strings in Give Settings Screen

### PR DESCRIPTION
**Pull request by ramiy**
Part o issue #876

@mathetos please review the changes.

### CODE REVIEW

File: includes/admin/class-give-settings.php [hunk 1]
```diff
@@ -249,7 +249,7 @@ public function give_settings( $active_tab ) {
 						),
 						array(
 							'name'    => esc_html__( 'Base Country', 'give' ),
-							'desc'    => esc_html__( 'Where does your site operate from?', 'give' ),
+							'desc'    => esc_html__( 'The country the site is operating from.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: Correction: "The country your site operates from."

File: includes/admin/class-give-settings.php [hunk 2]
```diff
@@ -262,15 +262,15 @@ public function give_settings( $active_tab ) {
...
+							'desc'    => esc_html__( 'The donation currency. Note that some payment gateways have currency restrictions.', 'give' ),
 							'id'      => 'currency',
 							'type'    => 'select',
 							'options' => give_get_currencies(),
 							'default' => 'USD',
 						),
 						array(
 							'name'    => esc_html__( 'Currency Position', 'give' ),
-							'desc'    => esc_html__( 'Choose the position of the currency sign.', 'give' ),
+							'desc'    => esc_html__( 'The position of the currency sign.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: Correction: "The position of the currency symbol." -- Reference: https://en.wikipedia.org/wiki/Currency_symbol

File: includes/admin/class-give-settings.php [hunk 3]
```diff
@@ -298,7 +298,7 @@ public function give_settings( $active_tab ) {
 						),
 						array(
 							'name'            => __( 'Number of Decimals', 'give' ),
-							'desc'            => __( 'This sets the number of decimal points shown in displayed prices.', 'give' ),
+							'desc'            => __( 'The number of decimal points in displayed prices.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: Correction: "The number of decimal points displayed in amounts."  -- "Prices" indicates products. But it's a Donation "amount".

File: includes/admin/class-give-settings.php [hunk 4]
```diff
@@ -329,13 +329,13 @@ public function give_settings( $active_tab ) {
 						),
 						array(
 							'name' => esc_html__( 'Enabled Gateways', 'give' ),
-							'desc' => esc_html__( 'Choose the payment gateways you would like enabled. You can drag-and-drop payment gateways to reorder them.', 'give' ),
+							'desc' => esc_html__( 'The supported payment gateways you would like enabled. You can drag-and-drop payment gateways to reorder them.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: Correction: "Enable your payment gateway. Can be ordered by dragging."

File: includes/admin/class-give-settings.php [hunk 5]
```diff
@@ -382,7 +382,7 @@ public function give_settings( $active_tab ) {
 						),
 						array(
 							'name' => esc_html__( 'Collect Billing Details', 'give' ),
-							'desc' => esc_html__( 'This option will enable the billing details section for offline donations. The fieldset will appear above the offline donation instructions. Note: You may customize this option per form as needed.', 'give' ),
+							'desc' => esc_html__( 'Enable billing details section for offline donations. The fieldset will appear above the offline donation instructions. Note: You may customize this option per form as needed.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: Correction: "Enable to request billing details for offline donations. Will appear above offline donation instructions. Can be enabled/disabled per form."

File: includes/admin/class-give-settings.php [hunk 6]
```diff
@@ -437,14 +437,14 @@ public function give_settings( $active_tab ) {
...
 							'id'   => 'enable_floatlabels',
 							'type' => 'checkbox'
 						),
 						array(
 							'name' => esc_html__( 'Disable Welcome Screen', 'give' ),
 							/* translators: %s: about page URL */
-							'desc' => sprintf( wp_kses( __( 'Enable this option if you would like to disable the Give Welcome screen every time Give is activated and/or updated. You can always access the <a href="%s">Welcome Screen</a> if you want in the future.', 'give' ), array( 'a' => array( 'href' => array() ) ) ), esc_url( admin_url( 'index.php?page=give-about' ) ) ),
+							'desc' => sprintf( wp_kses( __( 'Enable this option if you would like to disable the <a href="%s" target="_blank">Give Welcome screen</a> every time Give is activated and/or updated.', 'give' ), array( 'a' => array( 'href' => array(), 'target' => array() ) ) ), esc_url( admin_url( 'index.php?page=give-about' ) ) ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: I hate these types of settings. Enabling to disable things is just wrong. This setting really should be changed to say "Enable/Disable Welcome Screen" and be a radio box saying "Enable" and "Disable". Would be much more clear for the user and result in much fewer strings.

Sorry... will create a new issue about this. Need to review all the places where we do that.

File: includes/admin/class-give-settings.php [hunk 7]
```diff
@@ -501,13 +501,13 @@ public function give_settings( $active_tab ) {
...
 							'name' => esc_html__( 'Enable Form Categories', 'give' ),
-							'desc' => esc_html__( 'Check this option if you would like to categorize your donation forms. This option enables the form\'s category taxonomy.', 'give' ),
+							'desc' => esc_html__( 'Enable the form\'s category taxonomy to categorize donation forms.', 'give' ),
 							'id'   => 'enable_categories',
 							'type' => 'checkbox'
 						),
 						array(
 							'name' => esc_html__( 'Enable Form Tags', 'give' ),
-							'desc' => esc_html__( 'Check this option if you would like to tag your donation forms. This option enables the form\'s tag taxonomy.', 'give' ),
+							'desc' => esc_html__( 'Enable the form\'s tag taxonomy to tag donation forms.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: Correction: "Enables the 'Tag' taxonomy for all Give forms."

File: includes/admin/class-give-settings.php [hunk 8]
```diff
@@ -545,7 +545,7 @@ public function give_settings( $active_tab ) {
 						array(
 							'id'      => 'from_name',
 							'name'    => esc_html__( 'From Name', 'give' ),
-							'desc'    => esc_html__( 'The name donation receipts are said to come from. This should probably be your site or shop name.', 'give' ),
+							'desc'    => esc_html__( 'The name donation receipts are said to come from. This should probably be your site name.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: "The name which appears in the 'From' field in donation receipt emails."

File: includes/admin/class-give-settings.php [hunk 9]
```diff
@@ -651,7 +651,7 @@ public function give_settings( $active_tab ) {
 						array(
 							'id'      => 'session_lifetime',
 							'name'    => esc_html__( 'Session Lifetime', 'give' ),
-							'desc'    => esc_html__( 'Give will start a new session per user once they have donated. This option controls the lifetime a user\'s session is kept alive. An active session allows users to view donation receipts on your site without having to be logged in as long as they are using the same browser they used when donating.', 'give' ),
+							'desc'    => esc_html__( 'Control the lifetime a user\'s session is kept alive. Give will start a new session per user once they have donated. An active session allows users to view donation receipts on the site without having to be logged in.', 'give' ),
```
Comment (mathetos) [inline includes/admin/class-give-settings.php]: The length of time a user's session is kept alive. Give starts a new session per user upon donation. Sessions allow donors to view their donation receipts without being logged in.

**Comment by mathetos** [comment]
@ramiy Thanks for this submission. I reviewed and made new recommendations within the code itself.

Also, the Travis is failing on this for some reason. Perhaps your copy was a bit behind?

**Comment by ramiy** [comment]
@mathetos Ok, the strings updated.

You should probably review all the other strings in the settings screen.

Don't worry about Travis, it's not the string changes, it's the `apply_filters_deprecated()` function which introduced in wp4.6. When testing on older WP version it fails because the functions was not exist then.

**Comment by mathetos** [comment]
Looks good, thanks for this. I'll keep reviewing other settings to shorten more.

**Pull request merged**